rpcs3/rpcs3qt/update_manager.cpp

```diff
@@ -0,0 +1,513 @@
+﻿#include "stdafx.h"
+#include "update_manager.h"
+#include "rpcs3_version.h"
+#include "Utilities/StrUtil.h"
+#include "Crypto/sha256.h"
+
+#include <QMessageBox>
+
+#if defined(_WIN32)
+#include <windows.h>
+#include <CpuArch.h>
+#include <7z.h>
+#include <7zAlloc.h>
+#include <7zBuf.h>
+#include <7zCrc.h>
+#include <7zFile.h>
+#include <7zVersion.h>
+
+#define PATH_MAX MAX_PATH
+
+#else
+#include <unistd.h>
+#include <sys/stat.h>
+#endif
+
+update_manager::update_manager()
+{
+	m_manager.setRedirectPolicy(QNetworkRequest::NoLessSafeRedirectPolicy);
+}
+
+void update_manager::check_for_updates(bool automatic, QWidget* parent)
+{
+	if (QSslSocket::supportsSsl() == false)
+	{
+		LOG_ERROR(GENERAL, "Can't update! Please make sure your system supports SSL.");
+		if (!automatic)
+			QMessageBox::warning(parent, tr("Auto-updater"), tr("Can't update! Please make sure your system supports SSL."));
```

Comment:
Would it be possible to add an ifdef here to let windows users know that they need dll files?

Reply:
Sure but do we have an official link to provide(either wiki or for the dlls)?
I'd change it but not too sure what to put there exactly.

Reply:
Why do we not ship the dlls anyway?